File: hw/basurto/u3/hw24CompositePattern/hw24CompossitePattern.py/model/supervisor.py
from abc import ABC
from model.employee import Employee

class Supervisor(Employee, ABC):
    def __init__(self):
        super().__init__()
        self.direct_reports = []

    def state_name(self):
        super().state_name()
        for report in self.direct_reports:
            report.state_name()

    def add(self, employee):
        self.direct_reports.append(employee)

File: hw/basurto/u3/hw24CompositePattern/hw24CompossitePattern.js/model/Supervisor.js
import { Employee } from './Employee.js';

export class Supervisor extends Employee {
    constructor() {
        super();
        this.directReports = [];
    }

    stateName() {
        super.stateName();
        for (const report of this.directReports) {
            report.stateName();
        }
    }

    add(employee) {
        this.directReports.push(employee);
    }
}